app/models/identity.rb

```diff
@@ -2,6 +2,10 @@ class Identity < ActiveRecord::Base
   include NonNullUuid
 
   belongs_to :user
+  belongs_to :sp,
```

Comment:
The problem before is that `service_provider` is the string column name so we can't name the association the same thing otherwise we have problems (which resulted in the spec failures)

So I came up with `sp` as the name of the association, so you can do `identity.sp` like so:


So like:

```ruby
identity = Identity.new
service_provider = ServiceProvider.last

identity.sp = service_provider
# set implicitly by the line above
identity.service_provider == service_provider.issuer
# => true
```

The bummer is that none of the associations are used in our code, they're just for diagramming purposes.